cord/configs.py

```diff
@@ -61,9 +66,10 @@ def define_headers(self, data) -> Dict:
         return self._headers
 
     def __init__(self,
-                 resource_id: str = None,
-                 api_key: str = None,
-                 endpoint: str = CORD_ENDPOINT,
+                 resource_id: Optional[str] = None,
```

Comment:
I always recommend being precise about the types we are using. If someone sees the type `str` they should not expect it to ever be `None`.

Reply:
In general I also prefer being specific with types. However, in the specific case of keyword arguments with a default value of 'None' it is clear that the argument is Optional (and type checking systems pick up on it).

Reply:
Nice I didn't know this about the type checkers... super nice that PyCharm will give you the correct hint. I would still follow the Zen of Python and say "Explicit is better than implicit." :)

Maybe not a common situation in a public API, but if you ever were to change the default argument to a string (or remove a default argument), you'd have to change the type annotation. If you have it explicit there is no such cases to worry about.